### PR TITLE
Add middleware for running requests in an isolated registry

### PR DESCRIPTION
**Pull request by timothysu**
This PR adds Rack middleware to run each request in an isolated context registry.  It attempts to insert itself at the tip of the middleware stack when loaded through Rails.

Some bonus changes include bumping the ruby version and ignoring `/log/`.

prime: @jturkel

### CODE REVIEW

File: lib/rails_multitenant/middleware.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+module RailsMultitenant
+  class Middleware
+    def initialize(app)
+      @app = app
+    end
+
+    def call(env)
+      dup._call(env)
```
Comment (jturkel) [inline lib/rails_multitenant/middleware.rb]: Why do we need the `dup` + `_call`? Seems like we can simplify this and get rid of tracking the response in an instance variable.

Comment (timothysu) [reply lib/rails_multitenant/middleware.rb]: Misunderstanding around thread-safety here, re-read and updated!

**Comment by timothysu** [comment]
@jturkel confirming you're ok with file moves / renames before merging.  tested locally and it works!

**Comment by timothysu** [comment]
Will followup later if naming is not ideal.  Releasing `v0.7.0` for now.

**Comment by timothysu** [comment]
Accidentally merged with the default option - I've disabled every option besides "squash and merge" from the UI to prevent this for the future.

**Pull request merged**